tfhe/benches/integer/bench.rs

```diff
@@ -111,28 +118,59 @@ fn bench_server_key_binary_function_clean_inputs<F>(
     for (param, num_block, bit_size) in ParamsAndNumBlocksIter::default() {
         let param_name = param.name();
 
-        let bench_id = format!("{bench_name}::{param_name}::{bit_size}_bits");
-        bench_group.bench_function(&bench_id, |b| {
-            let (cks, sks) = KEY_CACHE.get_from_params(param, IntegerKeyKind::Radix);
+        let bench_id;
 
-            let encrypt_two_values = || {
-                let clear_0 = gen_random_u256(&mut rng);
-                let ct_0 = cks.encrypt_radix(clear_0, num_block);
+        // TODO do throughput only on clean inputs
 
-                let clear_1 = gen_random_u256(&mut rng);
-                let ct_1 = cks.encrypt_radix(clear_1, num_block);
+        match BENCH_TYPE.get().unwrap() {
+            BenchmarkType::Latency => {
+                bench_id = format!("{bench_name}::{param_name}::{bit_size}_bits");
+                bench_group.bench_function(&bench_id, |b| {
+                    let (cks, sks) = KEY_CACHE.get_from_params(param, IntegerKeyKind::Radix);
 
-                (ct_0, ct_1)
-            };
+                    let encrypt_two_values = || {
+                        let clear_0 = gen_random_u256(&mut rng);
+                        let ct_0 = cks.encrypt_radix(clear_0, num_block);
 
-            b.iter_batched(
-                encrypt_two_values,
-                |(mut ct_0, mut ct_1)| {
-                    binary_op(&sks, &mut ct_0, &mut ct_1);
-                },
-                criterion::BatchSize::SmallInput,
-            )
-        });
+                        let clear_1 = gen_random_u256(&mut rng);
+                        let ct_1 = cks.encrypt_radix(clear_1, num_block);
+
+                        (ct_0, ct_1)
+                    };
+
+                    b.iter_batched(
+                        encrypt_two_values,
+                        |(mut ct_0, mut ct_1)| {
+                            binary_op(&sks, &mut ct_0, &mut ct_1);
+                        },
+                        criterion::BatchSize::SmallInput,
+                    )
+                });
+            }
+            BenchmarkType::Throughput => {
+                bench_id = format!("{bench_name}::throughput::{param_name}::{bit_size}_bits");
+                let elements = throughput_num_threads();
+                bench_group.throughput(Throughput::Elements(elements));
+                bench_group.bench_function(&bench_id, |b| {
```

Comment:
unclear to me if this is the right way to call the throughput, did you confirm it works as expected ?

Reply:
Yes it works as expected, at least on my laptop. We'll need to double check on target machine to see if results are coherent.